src/filter.rs

```diff
@@ -13,10 +14,27 @@ pub fn filter(block_list: &Vec<String>, buf: &[u8]) -> String {
     });
 
     let result = result_itr.collect::<Vec<&str>>().join("\n");
-    
+
     return result;
 }
 
+pub fn filter_2(block_list: &[Bytes], data: &[u8]) -> Bytes {
+    let mut buffer = BytesMut::with_capacity(data.len());
+
+    'outer: for line in data.split(|x| x == &b'\n') {
+        for prefix in block_list {
+            if line.starts_with(prefix) {
+                continue 'outer;
+            }
```

Comment:
Current impl (as well as the initial one) runs in linear time of block list size.
Since this is main functionality it makes sense to optimize it. It is possible to implement it in constant time (of block list size), for example by using Trie (prefix tree) data structure

Reply:
On my first commit I was actually using BTreeSet to store the keys, but it required exact matches and was slower than the Vec + starts_with() approach, though I think it was due to other slow areas. I've read about Tries before, but I'm not experienced in them, so feel free to propose some samples if you have 😃 

I'm currently focusing on a way to remove the extra `BytesMut` allocation, which I think it's entirely possible, leaving the prefix match to be the only remaining bottleneck